src/Behat/Behat/Output/Node/EventListener/JUnit/JUnitOutlineStoreListener.php

```diff
@@ -54,12 +55,31 @@ public function __construct(SuitePrinter $suitePrinter)
     public function listenEvent(Formatter $formatter, Event $event, $eventName)
     {
         $this->captureOutlineOnBeforeOutlineEvent($event);
+        if ($event instanceof SuiteTested) {
+            setFormatterFileName($event, $formatter);
```

Comment:
I would also prefer having a method that returns the formater fileName and you simply assign the returned value of this method to `$formatter->fileName` (so you don't have to pass the `Formatter` object to your method)